### ISSUE TITLE
Regression: Failed inference on `TypedDict.get("key", [])`

### ISSUE THREAD

**Issue by pvandyken** [microsoft/pyright#5364]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**
As of pyright v1.1.106, it fails to infer the subtype of a list when calling `.get` IF the "gotten" key is of a sequence type other than `list`

**To Reproduce**
Python v3.9.7
pyright >=1.1.306,<1.1.316
(specifically tested on 1.1.306, 1.1.310, 1.1.312, 1.1.315)

Run pyright on the following file:

```py
from __future__ import annotations
from typing import Any, TypedDict

class typed_dict(TypedDict, total=False):
    dict_key: dict[str, Any]
    list_key: list[str]

def func(
    tdict: typed_dict,
    reg_dict: dict[str, dict[str, Any]],
):
    tdict.get("dict_key", [])
    #                     ~~
    #                     Argument Type is partially unknown

    tdict.get("list_key", [])

	# No error here... dict is correctly inferred as a sequence
    reg_dict.get("key", [])
```

```
/.../snakebids/test.py
  .../test.py:13:27 - error: Argument type is partially unknown
  Argument corresponds to parameter "default" in function "get"
  Argument type is "list[Unknown]" (reportUnknownArgumentType)
1 error, 0 warnings, 0 informations
```

**Expected behavior**
I expect correct inference of the generic type, as occurs for `reg_dict`

**VS Code extension or command-line**
Occurs in vscode extension and pip installed pyright

**Additional context**

**Comment by erictraut** [comment]
This change was intended, so I don't consider it a real regression. However, I agree the current behavior is inconsistent. The `dict.get` and `TypedDict.get` should behave the same when a default value is passed. I can make the argument that that pyright should evaluate `[]` as `list[Unknown]` in both cases. There is some special-case code that evaluates `[]` as `list[Any]` rather than `list[Unknown]` in the case of the `dict.get` call. I'm reluctant to change this behavior (even though I think it would be more correct) because it would be unwelcome by many pyright users. I've therefore decided to make the special case apply to the `TypedDict.get` method as well. Now they both produce `list[Any]`, which is what you were hoping to see.

This will be included in the next release.

**Comment by pvandyken** [comment]
> I can make the argument that that pyright should evaluate `[]` as `list[Unknown]` in both cases.

Right. I use this pattern probably exclusively in an iterator context, where the exact container identity is irrelevant, but I can see how if you did a simple assignment there could be problems. But thanks for fixing the inconsistency

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.316, which I just published. It will also be included in this week's insiders build of pylance.
